pkg/models/context.go

```diff
@@ -0,0 +1,86 @@
+package models
+
+import (
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/macaron.v1"
+
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/grafana/grafana/pkg/services/session"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Context struct {
```

Comment:
I think we should come up with another name for this struct. It conflicts with golangs internal [Context package](https://golang.org/pkg/context). While we might know the difference, I think its confusing for external developers. 

How about `RequestContext` or `ReqContext`? 

Renaming things in GO is kinda easy using vscode go plugin or Goland (https://www.jetbrains.com/go)